nemo_skills/pipeline/start_server.py

```diff
@@ -38,6 +48,51 @@ def get_gradio_chat_cmd(model, server_type, extra_args):
     return cmd
 
 
+def create_server_tunnel(exp: Experiment, server_port: int, wait_interval: int = 10):
+    ## Assumes first job in experiment correspond to the server.
+    job: Job = exp.jobs[0]
+
+    if not isinstance(job.executor.tunnel, SSHTunnel):
+        LOG.warning("Not using an SSH tunnel, skipping server tunnel.")
+        return
+
+    LOG.info("Waiting for server job to start...")
+    while job.status(exp._runner) is not AppState.RUNNING:
+        time.sleep(wait_interval)
+
+    try:
+        _, _, path_str = job.handle.partition("://")
+        path = path_str.split("/")
+        app_id = path[1]
+    except Exception as e:
+        LOG.exception("Unable to get job ID for server tunnel.")
+        LOG.exception(e)
+        return
+
+    job_node_list_cmd: RunResult = job.executor.tunnel.run(f"scontrol show job {app_id} | grep -E '\s+NodeList'")
+    if job_node_list_cmd.return_code == 0:
+        _, job_node_list = job_node_list_cmd.stdout.strip().split("=", 1)
+        job_node_list = job_node_list.split(",")
+    else:
+        LOG.exception(f"Failed to get node list. {job_node_list_cmd.stderr}")
+        return
+
+    ssh_tunnel_args = [
+        "ssh",
+        "-N",
+        "-o",
+        "StrictHostKeyChecking=accept-new",
+        "-J",
+        f"{job.executor.tunnel.user}@{job.executor.tunnel.host}",
+        job_node_list[0],
+        "-L",
+        f"{server_port}:localhost:{server_port}",
+    ]
+    LOG.info(f"SSH tunnel command: {' '.join(ssh_tunnel_args)}")
```

Comment:
would be nice to print a message for the final host/port for the server to access locally. host is always localhost, but port can be random, so would be nice to print here so that it's easy to know where to connect. Can even add what exactly should be set in ++server_address argument e.g. Alternatively we can accept server/sandbox local port to map to

Reply:
I've added these details to the docs.